src/views/Login.js

```diff
@@ -0,0 +1,163 @@
+
+export class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errors: {}
+    };
+    this.handleOnChange = this.handleOnChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleNavigation = this.handleNavigation.bind(this);
+  }
+
+  handleOnChange(e) {
+    this.setState({
+      errors: {}
+    });
+    const { name, value } = e.target;
+    const { handleTextInput: textInput } = this.props;
+    textInput(name, value);
+  }
+
+  handleSubmit() {
+    const { email, password, handleSignIn: signIn } = this.props;
+    const errors = Validator.formData({ email, password });
+    if (!isEmpty(errors)) {
+      return this.setState({
+        errors: {
+          ...errors
+        }
+      });
+    }
+    signIn({ email, password });
+  }
+
+  handleNavigation() {
+    const { message } = this.props;
+    return message;
+  }
+
+  render() {
+    const { email, password, message, isSubmitting } = this.props;
+    const { errors } = this.state;
+    return (
+      <div className="auth" data-test="login">
+        <div className="row">
+          <div className="col-md-6 hide-sm">
+            <div className="auth-left" data-test="auth-left">
+              <div className="logo">
+                <img
+                  src={require("../assets/img/quill-drawing-a-line.svg")}
+                  alt="logo"
+                  className="logo"
+                  data-test="logo"
+                />
+              </div>
+              <p>Join the community of like minded author, today!</p>
+              <BasicButton
+                className="btn btn-block"
+                onClick={() => this.handleNavigation()}
+                title="Sign Up"
+                data-test="nav-link"
+              />
+            </div>
+          </div>
+          <div className="col-md-6 col-sm-12">
+            <div className="auth-right" data-test="auth-right">
+              <div className="auth-form">
+                <div className="hide-md logo">
+                  <img
+                    src={require("../assets/img/quill-drawing-a-line.svg")}
+                    alt="logo"
+                    className="logo"
+                    data-test="logo"
+                  />
+                </div>
+                <p className="pg-title">Sign in with</p>
+                <div className="socials" data-test="socials">
+                  <SocialButton iconName="fb" alt="fb" />
+                  <SocialButton iconName="twitter" alt="twitter" />
+                  <SocialButton iconName="google-plus" alt="G" />
+                </div>
+                <p className="pg-title">Or</p>
+                <form data-test="login-form">
+                  <TextInput
+                    type="email"
+                    name="email"
+                    id="auth-email"
+                    placeholder="Email"
+                    onChange={e => this.handleOnChange(e)}
```

Comment:
Use arrow functions hence no need for this context

Reply:
I think, with arrow functions, we will still have to use this context. They are properties of a component. Removing ```this``` result in ```undefined``` error.